Library/Homebrew/download_strategy.rb

```diff
@@ -532,6 +532,63 @@ def _fetch
   end
 end
 
+# GitHubReleaseDownloadStrategy downloads tarballs from GitHub Release assets.
+# To use it, add ":using => GitHubReleaseDownloadStrategy" to the URL section
+# of your formula. This download strategy uses GitHub access tokens (in the
+# environment variables HOMEBREW_GITHUB_API_TOKEN) to sign the request.
+# This strategy is suitable for corporate use just like S3DownloadStrategy,
+# because it lets you use a private GttHub repository for internal distribution.
+# It works with public one, but in that case simply use CurlDownloadStrategy.
+class GitHubReleaseDownloadStrategy < CurlDownloadStrategy
+  require "utils/formatter"
+  require 'utils/github'
+
+  def initialize(name, resource)
+    super
+
+    @github_token = ENV["HOMEBREW_GITHUB_API_TOKEN"]
+    raise CurlDownloadStrategyError, "Environmental variable HOMEBREW_GITHUB_API_TOKEN is required." unless @github_token
```

Comment:
Also, you probably need to handle the case where the token doesn't have e.g. private repository access.